source/Octopus.Versioning/Maven/MavenVersion.cs

```diff
@@ -78,15 +79,7 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            unchecked
-            {
-                var hashCode = Major;
-                hashCode = (hashCode * 397) ^ Minor;
-                hashCode = (hashCode * 397) ^ Patch;
-                hashCode = (hashCode * 397) ^ Revision;
-                hashCode = (hashCode * 397) ^ (ReleaseLabels != null ? ReleaseLabels.GetHashCode() : 0);
```

Comment:
`ReleaseLabels.GetHashCode()` -> this part was breaking MavenVersion's hashcode (not entirely sure why).

Reply:
See https://github.com/OctopusDeploy/Versioning/pull/41#discussion_r804351048